### PR TITLE
Simplify x == x

### PR DESCRIPTION
**Pull request by shoaibkamil**
For both boolean and numeric values of x.  An attempt to address #3718 in the simplest manner possible.

### CODE REVIEW

File: src/Simplify_EQ.cpp [hunk 1]
```diff
@@ -39,6 +42,12 @@ Expr Simplify::visit(const EQ *op, ExprInfo *bounds) {
     Expr delta = mutate(op->a - op->b, &delta_bounds);
     const int lanes = op->type.lanes();
 
+    // If the bounds are 0, then it's just x == x
```
Comment (abadams) [inline src/Simplify_EQ.cpp]: I don't think this actually covers floating point x == x, as the bounds won't be defined. I think the missing rewrite (in addition to the one above), is actually 0 == 0

Comment (abadams) [reply src/Simplify_EQ.cpp]: Though written like that it's obviously not going to be a matcher. Maybe we should just say if (is_zero(delta)) for the condition here.

Comment (shoaibkamil) [reply src/Simplify_EQ.cpp]: Ah right, I forgot bounds won't be defined for floating point.

**Comment by slomp** [comment]
I remember auditing that piece of code many weeks ago when I first encountered the problem, and also completely missed that case :(

**Pull request merged**